examples/project1/CMakeLists.txt

```
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 
 project(project1 C Fortran)
 
+string(REPLACE "--generate-object-code" "" CMAKE_Fortran_COMPILE_OBJECT "${CMAKE_Fortran_COMPILE_OBJECT}")
```

Please add a comment here explaining why the line is needed.

Once a CMake is released without this hard-coded flag, hopefully 3.31.4, then this can be dropped, e.g.

```suggestion
if(CMAKE_VERSION VERSION_LESS 3.31.4)
  # CMake 3.31.0 through 3.31.3 hard-coded this flag.
  # It does not work with modules, so remove it.
  string(REPLACE "--generate-object-code" "" CMAKE_Fortran_COMPILE_OBJECT "${CMAKE_Fortran_COMPILE_OBJECT}")
endif()
```

---

This suggestion seems to have been dropped by one of the force pushes.

---

I am sorry. I had made this change through the GitHub web interface and it seems I might have forgotten to pull that commit before I force pushed. I reintroduced the change in [2868b0a](https://github.com/lfortran/lfortran/pull/5945/commits/2868b0af4aca55f33c94fda9cf04c2ba44d72f0a).